File: abc155/b/c++/b.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
  int n;
  cin >> n;
  vector<int> a(n);
  for(int i = 0; i < n; i++){
    cin >> a[i];
  }

  for(int j = 0; j < n; j++){
    if(a[j] % 2 == 0 && a[j] % 3 != 0 && a[j] % 5 != 0){
      cout << "DENIED" << endl;
      return 0;
    }
  }
  cout << "APPROVED" << endl;

  return 0;
}

File: abc155/b/b.ts
import * as fs from "fs";

function main(){
  const input = fs.readFileSync("/dev/stdin", "utf-8").trim().split(/\s+/);
  const n = parseInt(input[0], 10);
  const a = input.slice(1).map(Number);

  for(let i = 0; i < n; i++){
    if(a[i] % 2 == 0 && a[i] % 3 != 0 && a[i] % 5 != 0){
      console.log("DENIED");
      return 0;
    }
  }
  console.log("APPROVED");
}

main();
